test/integration/transactions/transactions.prose.test.ts

```diff
@@ -0,0 +1,73 @@
+import { expect } from 'chai';
+import * as semver from 'semver';
+
+import { type MongoClient } from '../../mongodb';
+
+const metadata: MongoDBMetadataUI = {
+  requires: {
+    topology: '!single'
+  }
+};
+
+describe('Transactions Spec Prose', function () {
+  let client: MongoClient;
+  const started = [];
+
+  beforeEach(async function () {
+    if (
+      semver.satisfies(this.configuration.version, '<=4.2') &&
+      this.configuration.topologyType === 'Sharded'
+    ) {
+      if (this.currentTest) {
+        this.currentTest.skipReason =
+          'Transactions on sharded clusters are only supported after 4.2';
+        this.skip();
+      }
+    }
+    started.length = 0;
+    client = this.configuration.newClient({}, { monitorCommands: true });
+
+    await client
+      .db()
+      .collection('txn-test')
+      .drop()
+      .catch(() => null);
+    await client.db().createCollection('txn-test');
+
+    client.on('commandStarted', ev => started.push(ev));
+  });
+
+  afterEach(async function () {
+    await client
+      ?.db()
+      .collection('txn-test')
+      .drop()
+      .catch(() => null);
+    await client?.close();
+  });
+
+  describe('Options Inside Transaction', function () {
+    it(
+      '1.0 Write concern not inherited from collection object inside transaction.',
+      metadata,
+      async () => {
+        await client.withSession(async session => {
+          session.startTransaction();
+
+          const collection = client.db().collection('txn-test', { writeConcern: { w: 0 } });
+
+          await collection.insertOne({ n: 1 }, { session });
+
+          await session.commitTransaction();
+        });
+
+        const insertStarted = started.find(ev => ev.commandName === 'insert');
+        expect(insertStarted).to.not.have.nested.property('command.writeConcern');
+
```

Comment:
Are we missing the following check?

> Ensure the document was inserted and no error was thrown from the transaction.

Reply:
No, its just by nature of the insert / commitTransaction not throwing. Checking the command started is sort of unreachable, if the WC is set the commitTxn throws, but its a reasonable double-check

Reply:
Python checks that the result of the insert is an inserted id.  We could also query the collection after the transaction for our doc - either would satisfy the "confirm the document was inserted".

This is important because w: 0 doesn't throw but might not insert.

Reply:
Gotcha, will do

Reply:
added